ui/desktop/src/hooks/useRecipeManager.ts

```diff
@@ -117,16 +126,19 @@ export const useRecipeManager = (messages: Message[], locationState?: LocationSt
 
     // Otherwise, we are waiting for parameters, so the input should be empty.
     return '';
-  }, [recipeConfig, recipeParameters]);
+  }, [recipeConfig, recipeParameters, messages.length]);
 
   // Handle parameter submission
   const handleParameterSubmit = async (inputValues: Record<string, string>) => {
-    setRecipeParameters(inputValues);
+    // Store parameters in chat context instead of local state
+    if (chatContext?.setRecipeParameters) {
+      chatContext.setRecipeParameters(inputValues);
```

Comment:
in what case is chatContext undefined here? can we use the type system to make sure it is not?